File: app/src/main/java/com/example/bancodedadossqlite/EditTaskActivity.java
package com.example.bancodedadossqlite;

import android.app.Activity;
import android.content.Intent;
import android.widget.Toolbar;

import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textfield.TextInputLayout;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.Extra;
import org.androidannotations.annotations.ViewById;

@EActivity(R.layout.activity_addedit)
public class EditTaskActivity extends Activity {

    private Toolbar toolbar;

    @ViewById
    TextInputLayout inputTask;

    @ViewById
    TextInputLayout inputCardLink;

    @Extra("Task")
    Task task;

    @AfterViews
    void afterViews(){
        inputTask.getEditText().setText(task.getTask());
        inputCardLink.getEditText().setText(task.getCardLink());
    }

    @Click(R.id.buttonConfirm)
    void onConfirmClick() {
        task.setTask( inputTask.getEditText().getText().toString() );
        task.setCardLink( inputCardLink.getEditText().getText().toString() );

        Intent intent = getIntent();
        intent.putExtra("newTask", task);
        setResult(RESULT_OK,intent);

        onBackPressed();
    }



}

File: app/src/main/java/com/example/bancodedadossqlite/EditTaskActivity.kt
package com.example.bancodedadossqlite

import android.app.Activity
import android.widget.Toolbar
import com.google.android.material.textfield.TextInputLayout
import org.androidannotations.annotations.*

@EActivity(R.layout.activity_addedit)
open class EditTaskActivity : Activity() {

    @ViewById
    lateinit var inputTask: TextInputLayout

    @ViewById
    lateinit var inputCardLink: TextInputLayout

    @Extra("Task")
    lateinit var task: Task

    @AfterViews
    fun afterViews() {
        inputTask.editText?.setText(task.task)
        inputCardLink.editText?.setText(task.cardLink)
    }

    @Click(R.id.buttonConfirm)
    fun onConfirmClick() {
        task.task = inputTask.editText?.text.toString()
        task.cardLink = inputCardLink.editText?.text.toString()
        intent.putExtra("newTask", task)
        setResult(RESULT_OK, intent)
        finish()
    }

    override fun onBackPressed() {
        setResult(RESULT_CANCELED, intent)
        super.onBackPressed()
    }
}